langgraph/src/pregel/index.ts

```diff
@@ -347,21 +371,30 @@ export class Pregel
 
       // eslint-disable-next-line @typescript-eslint/no-explicit-any
       const pendingWrites: Array<[string, any]> = [];
+      // A copy of the checkpoint is created because `checkpoint` is defined with `let`.
+      // `checkpoint` can be mutated during loop execution and when used in a function,
+      // may cause unintended consequences.
+      const checkpointCopy = copyCheckpoint(checkpoint);
```

Comment:
This might be temporary. When `stream()` is reimplemented, this code might be removed.